### DISCUSSION TITLE
[QUESTION] preferences: search language: default language

### DISCUSSION THREAD

**Discussion by Golddouble** [searx/searx#3102] [General]
I use searx via https://suche.honigdachse.de/

In preferences in "general tab" there is a setting "search language", where we can choose the language. There we can choose "default language". I do not understand "default language". What exactly does the setting "default language" make? For example when I use the engine "google".

![ksnip_20211222-181042](https://user-images.githubusercontent.com/13977359/147137404-98278d90-499b-4a9d-a0af-4bdb38a23167.jpg)

Would appreciate an answer. Thank you.

**Comment by MarcAbonce** [comment]
Yeah, that name is confusing.
It actually refers to the engine's "default language". That is, if you select that option searx will not send a language parameter with your query but will instead let the engine guess the language or do whatever it wants.
So it's more like a "None" option.

**Comment by Golddouble** [comment]
Thank you. Thank you @MarcAbonce 

When I search for officially with
https://suche.honigdachse.de/?preferences=eJx1VUuP3DYM_jX1xdigaQ49-VC0KBIgwAbZaa8GLXFs1pLoUPLMOr--lGfGI082BxsQxcfHjw8ZSNizEMamx4ACrnIQ-hl6bMDpgQ04bDBUMCc27CeHCZuKvGq0k_Dr0hxkxspjGtg2X55fDlWEI0YEMUPza5UG9NhwNCCVYJxdii2HNuC5TdA1f4OLWFmmVi_ZnVCaaOhpmLt36VytZk8xLYrBcU-GLZ4qsCcIBm17DbICsBShcyrE0FPQfOxsxvz13LbX3H757c-YQNKUwRdCFD6TLSVhAWjbNcuoRzOLYDBLqULhRJZ4jm3r50hGRT1z77C-mBXmmBbPwSms0kFPyUGnakkPZxqpYx7jDldidjtJjzgm8qv3bDaSGSHq6UhuDXW0wjmV2_kKaXKw1J5PlC0VN3K-vPCXSa3z7w4lEQeQ5UdGNtOOQv_j9QMRa1SYpgJfzvPbzGnHBE8YBCcu9CJ-D-BLpQUG3pUy8bhw4jjwCOFuGSZfcKoNFSntMpG5W3r0NwoplHn11GsbQUw7C7SWVBLZELjaoyVQ8bczhJ3eURDryMd0BsHakqBJnGlcA71_LVpigE4g_x4LuSFxpBqy1Nl9pF0XOO5iwndyT2Frxc0cprH2JMJyp6aj1OlEYLra5aJ3KP31eEtzA3kt48Vn4fuB-VvVM99T5qbEasmk77zvfL949EpMnQRCdLqDdsOX_VhIOzcBtJJv1aSYjQta3SyKNf9vLcDLw7wfhcJIYIpU8yBos9R7ot9_-PD7a5G4MeYpnQqBtX1t8UiB8sjEN-BEM7ADha5rTVcIruHBQy4vPhB_Bb31wlvjtMr2nO8ynhAlzR3-vFyb3PN_uk4emY88i8G9tHtjFaxNVWHYb90L1E37vpYnN6tGbD5yTDrZqPte81n5-Jf8k6MR24HTiEuG9imsnKKyJuxy0GfdEe0fxqAS89fzJzU_iypsADb_EIfrQa0OAtrv0v7z9bNa6AJEUekLumOrIVg8rHVbZbkF8tNkLm_ios-L0wHWu4-Hw5eXLaLSiRrof-hGuiU=&q=%s

Then I get  4,330,000,000 results.

You see I only have activated "google" engine and "default"

But when I search 
officailly 
with https://www.google.com/search?q=officially
I get 413’000’000 results. 

Why do I get about 1000 times less results with google directly? Strange? My expectation was that I will get less results with searx than with google and not the other way round. Very confusing.

**Comment by jsfan3** [comment]
I have a question in this regard.

My `/etc/searx/settings.yml` file contains:
`default_lang : "en-IT" # Default search language - leave blank to detect from browser information or use codes from 'languages.py'`

I want my searches, by default, to be valid in any language. How can I modify settings.yml so that the "default language" option is on by default, as understood in this discussion?

So far, whenever I've used SearX, I've often had to specify the search language manually (I usually do searches in multiple languages). If it were automatically detected, that would be a great thing.

**Reply by 0xhtml** [reply]
You can set `default_lang` to `all` e.g.
```yml
default_lang : "all"
```
. That is equivalent to selecting "default language".
